src/module_wrap.cc

```diff
@@ -702,35 +715,43 @@ static MaybeLocal<Promise> ImportModuleDynamically(
   Environment* env = Environment::GetCurrent(context);
   v8::EscapableHandleScope handle_scope(iso);
 
-  if (env->context() != context) {
-    auto maybe_resolver = Promise::Resolver::New(context);
-    Local<Promise::Resolver> resolver;
-    if (maybe_resolver.ToLocal(&resolver)) {
-      // TODO(jkrems): Turn into proper error object w/ code
-      Local<Value> error = v8::Exception::Error(
-        OneByteString(iso, "import() called outside of main context"));
-      if (resolver->Reject(context, error).IsJust()) {
-        return handle_scope.Escape(resolver.As<Promise>());
-      }
-    }
-    return MaybeLocal<Promise>();
-  }
-
   Local<Function> import_callback =
     env->host_import_module_dynamically_callback();
+
+  Local<PrimitiveArray> options = referrer->GetHostDefinedOptions();
+
+  Local<Value> object;
+
+  int type = options->Get(0).As<Number>()->Int32Value(context).ToChecked();
```

Comment:
The first item in https://github.com/nodejs/node/pull/22381#discussion_r212804782 doesn't seem to be resolved here. As a bonus, can we use a system similar to https://github.com/nodejs/node/blob/master/src/node_context_data.h here?